### PR TITLE
CI: Change push event trigger to run checks only on protected branches

### PR DESCRIPTION
**Pull request by hse-root**
Docs check now only runs when any of the files under docs are touched

Signed-off-by: hse-devops <hse_devops@micron.com>

### CODE REVIEW

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+name: Docs
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened, edited]
+    paths:
+    - 'docs/**'
```
Comment (tristan957) [inline .github/workflows/docs.yml:7]: This also needs include/**

File: .github/workflows/docs.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      image: fedora:latest
+      options: --privileged  -v /usr/src:/usr/src -v /lib/modules:/lib/modules
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Initialize
+      run: |
+        echo "$HOME/.local/bin" >> $GITHUB_PATH
+        sudo dnf install -y git poetry python-unversioned-command pkg-config java-1.8.0-openjdk-devel libmicrohttpd-devel userspace-rcu-devel libyaml-devel lz4-devel libbsd-devel libcurl-devel doxygen
```
Comment (tristan957) [inline .github/workflows/docs.yml:22]: You don't need the jdk

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -2,6 +2,9 @@ name: CI
 
 on:
   push:
+    branches:
+    - master
+    - v2.0
```
Comment (tristan957) [inline .github/workflows/ci.yaml:7]: You should wildcard the v2.0 to v*.*

File: .github/workflows/docs.yml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+jobs:
+  docs:
+    runs-on: ubuntu-latest
+    container:
+      image: fedora:latest
+      options: --privileged  -v /usr/src:/usr/src -v /lib/modules:/lib/modules
+
+    steps:
+    - uses: actions/checkout@v2
```
Comment (tristan957) [inline .github/workflows/docs.yml:17]: Indent the - for all of these.

**Review by tristan957 — APPROVED** [review]

File: .github/workflows/docs.yml (hunk 1 above)
Comment (tristan957) [reply .github/workflows/docs.yml:7]: Make the glob only care about .h files

File: .github/workflows/docs.yml [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+      run: |
+        echo "$HOME/.local/bin" >> $GITHUB_PATH
+        sudo dnf install -y git poetry python-unversioned-command pkg-config java-1.8.0-openjdk-devel libmicrohttpd-devel userspace-rcu-devel libyaml-devel lz4-devel libbsd-devel libcurl-devel doxygen
+        sudo dnf install -y xz mongo-c-driver libbson openssl-devel cyrus-sasl-devel ncurses-devel cmake make automake gcc gcc-c++ kernel-devel
+        sudo dnf install -y python-pip python-wheel python3-devel libxml2-devel libxslt-devel
+        poetry install
+
+    - name: Setup
+      run: |
+        poetry run meson builddir --fatal-meson-warnings -Dwerror=true -Ddocs=true
```
Comment (tristan957) [inline .github/workflows/docs.yml:29]: -Dtests=false -Dtools=disabled -Dbindings=none -Dcli=false -Dsamples=false

File: .github/workflows/docs.yml (hunk 1 above)
Comment (tristan957) [reply .github/workflows/docs.yml:7]: Indent the array

File: .github/workflows/ci.yaml (hunk 3 above)
Comment (hse-root) [reply .github/workflows/ci.yaml:7]: I am going to make that v2.*, we can always add v3.* when we get there. I dont want this running on v1.x.

**Review by leroyjvargis — APPROVED** [review]

**Pull request merged**